### PR TITLE
Rename VaultEndpoint to VaultUri

### LINKED ISSUE #8332: Rename vaultEndpoint to vaultUri/vaultUrl (idiomatic)
After discussions with the arch board, we have decided to rename `vaultEndpoint` (parameters, properties, etc.) to `vaultUri` or `vaultUrl` (idiomatic). This is true for all clients and models where this exists across:

- [ ] Secrets
- [ ] Keys
- [ ] Certificates

### PR DESCRIPTION
**Pull request by heaths**
Fixes #8332

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs [hunk 1]
```diff
@@ -66,7 +66,10 @@ public CertificateClient(Uri vaultUri, TokenCredential credential, CertificateCl
             _pipeline = new KeyVaultPipeline(vaultUri, options.GetVersionString(), pipeline, new ClientDiagnostics(options));
         }
 
-        // Certificates API
+        /// <summary>
+        /// Gets the <see cref="Uri"/> of the vault used to create this instance of the <see cref="CertificateClient"/>.
+        /// </summary>
+        public Uri VaultUri => _pipeline.VaultUri;
```
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs]: virtual for mocking.

Comment (pakrym) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs]: in all the clients.

**Review by pakrym — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs]: Okay. FWIW, our current instrumenting functionality for recording doesn't mock properties, but I can understand how third-party code might.

**Review by tg-msft — APPROVED** [review]
Looks nice

**Pull request merged**